### PR TITLE
CU-1wzgvc6 sample of cross chain transaction for SELL (DEX/liqudation)

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

- make benches parts of tests
- fixes liqudatin interface
- document sometings
- made binary and typed call to liqudation

### CODE REVIEW

File: frame/dutch-auction/log.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+warning: Patch `polkadot v0.9.17 (https://github.com/composableFi/polkadot?branch=release-v0.9.17#1db48fed)` was not used in the crate graph.
```
Comment (vivekvpandya) [inline frame/dutch-auction/log.txt:1]: Is this file required?

File: integration-tests/local-integration-tests/src/transact_calls.rs [hunk 2]
```diff
@@ -1,6 +1,176 @@
...
-// liqudation call
-// lending call
+//! Testing custom XCMP transactions call
+
+use composable_traits::defi::Sell;
+use frame_system::EventRecord;
+use primitives::currency::CurrencyId;
+use support::{assert_ok, log};
+use xcm_emulator::TestExt;
+// /use xcm_executor::{traits::Convert, XcmExecutor};
```
Comment (vivekvpandya) [inline integration-tests/local-integration-tests/src/transact_calls.rs:8]: if not required then remove this.

File: frame/bonded-finance/src/lib.rs [hunk 3]
```diff
@@ -51,6 +51,19 @@
 //! Alice may cancel offer and prevent new bonds on offer, she gets her native tokens back.
 //! All existing vesting periods continue to be executed.
 //!
+//! ## Use cases
+//!
+//! - Staking. User locks amount for some period of time, gets reward in the end or vested. Stake
+//!   returns to user in the end.
+//! - Vesting. User offers amount and ensure that recepients has 'ticket' tokens to bond to get
+//!   reward.
+//! - Sell/Exchange/Trae. User makes bond offer, other users takes offer in exchange for other
```
Comment (vivekvpandya) [inline frame/bonded-finance/src/lib.rs:60]: `Trae` -> `Trade` ?

File: frame/dutch-auction/src/weights.rs [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#![allow(unused_parens, unused_imports, clippy::unnecessary_cast)]
+//#![allow(unused_parens, unused_imports, clippy::unnecessary_cast)]
```
Comment (vivekvpandya) [inline frame/dutch-auction/src/weights.rs:1]: Is this required now?

File: frame/dutch-auction/src/lib.rs [hunk 5]
```diff
@@ -119,29 +117,10 @@ pub mod pallet {
...
-				Balance = <Self as DeFiComposableConfig>::Balance,
-			>;
-		type WeightInfo: WeightInfo;
-		type PalletId: Get<PalletId>;
-		type NativeCurrency: NativeTransfer<Self::AccountId, Balance = Self::Balance>
-			+ Currency<Self::AccountId>;
-		/// Convert a weight value into a deductible fee based on the currency type.
-		type WeightToFee: WeightToFeePolynomial<Balance = Self::Balance>;
+
+		/// ED taken to create poisiton. Part of if returned when position is liqudated.
```
Comment (andor0) [inline frame/dutch-auction/src/lib.rs:121]:
```
-ED taken to create poisiton.
+ED taken to create position.
```

File: frame/liquidations/Cargo.toml [hunk 6]
```diff
@@ -19,6 +19,7 @@ version = "2.0.0"
 frame-benchmarking = { default-features = false, optional = true, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.17" }
 frame-support = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.17" }
 frame-system = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.17" }
+pallet-dutch-auction = { default-features = false, optional = true, path = "../dutch-auction" }
```
Comment (andor0) [inline frame/liquidations/Cargo.toml:22]: maybe should it move in `dev-dependencies` section?

File: integration-tests/local-integration-tests/src/transact_calls.rs [hunk 7]
```diff
@@ -1,6 +1,175 @@
...
+			WithdrawAsset(assets.clone().into()), /* withdrow native on target chain from origin
+			                                       * account */
+			BuyExecution {
+				// pay for origin account
+				fees: assets,
+				weight_limit: Unlimited,
+			},
+			Transact {
+				origin_type: OriginKind::SovereignAccount,
+				require_weight_at_most: UnitWeightCost::get() * 2,
```
Comment (vivekvpandya) [inline integration-tests/local-integration-tests/src/transact_calls.rs:116]: how is this `require_weight_at_most` is determined for this test?

File: integration-tests/local-integration-tests/src/transact_calls.rs [hunk 8]
```diff
@@ -1,6 +1,175 @@
...
+	});
+
+	Sibling::execute_with(|| {
+		let assets: MultiAsset = (
+			(Parent, X2(Parachain(THIS_PARA_ID), GeneralKey(this_native_asset.encode()))),
+			some_native_amount,
+		)
+			.into();
+		let xcm = vec![
+			WithdrawAsset(assets.clone().into()), /* withdrow native on target chain from origin
```
Comment (vivekvpandya) [inline integration-tests/local-integration-tests/src/transact_calls.rs:107]: minor typo: `withdrow` -> `withdraw`

File: frame/bonded-finance/src/lib.rs [hunk 9]
```diff
@@ -51,6 +51,19 @@
...
+//! - Staking. User locks amount for some period of time, gets reward in the end or vested. Stake
+//!   returns to user in the end.
+//! - Vesting. User offers amount and ensure that recepients has 'ticket' tokens to bond to get
+//!   reward.
+//! - Sell/Exchange/Trade. User makes bond offer, other users takes offer in exchange for other
+//!   amount..
+//!
+//! ## Concerns
+//!
+//! Protocol is not protected from sniper bots, whales and other attackers.
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply frame/bonded-finance/src/lib.rs:65]: amount of offer should depend on time and amount of already taken, or could be forced for vesting for some period

File: frame/composable-traits/src/liquidation.rs [hunk 10]
```diff
@@ -17,3 +19,21 @@ pub trait Liquidation: DeFiEngine {
 		configuration: sp_std::vec::Vec<Self::LiquidationStrategyId>,
 	) -> Result<Self::OrderId, DispatchError>;
 }
+
+/// generic transaction which can target any pallet and any method in any parachain (local or
+/// remote)
+/// so it must be encoded in format with widest possible values to incoroporate some chains we do
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply frame/composable-traits/src/liquidation.rs:25]: not sure i get it? spelling?

**Review by vivekvpandya — APPROVED** [review]

**Review by andor0 — APPROVED** [review]
